controllers/repo_manager/precheck.go

```diff
@@ -183,13 +183,6 @@ func checkImmutableFields(ctx context.Context, r *RepoManagerReconciler, pulp *r
 	// Checking immutable fields update
 	immutableFields := []immutableField{
 		{FieldName: "DeploymentType", FieldPath: repomanagerpulpprojectorgv1beta2.PulpSpec{}},
-		{FieldName: "ObjectStorageAzureSecret", FieldPath: repomanagerpulpprojectorgv1beta2.PulpSpec{}},
-		{FieldName: "ObjectStorageS3Secret", FieldPath: repomanagerpulpprojectorgv1beta2.PulpSpec{}},
-		{FieldName: "DBFieldsEncryptionSecret", FieldPath: repomanagerpulpprojectorgv1beta2.PulpSpec{}},
```

Comment:
Can we keep `DBFieldsEncryptionSecret` immutable for now? If a user changes it without properly following the key rotation (https://pulpproject.org/pulpcore/docs/admin/reference/settings/#db_encryption_key) then Pulp will become unusable and you will have to drop your database and start over. Until we can add this fancy rotation logic to the operator I think it is best to leave it immutable.

Reply:
That would be possible. However admins _can_ already shoot themselves in the foot when changing the value within the secret. And if someone wants to change to another secret, referencing the _same_ secret value, they could not do so.

So IMO keeping it immutable would just be snake oil

Reply:
That makes sense. I would think of adding it [here](https://pulpproject.org/pulp-operator/docs/admin/guides/configurations/secrets/#pulp-db-fields-encryption) with a warning box as well as changing the docs on the parameter for kubebuilder generation (and work with some caps in the latter case).

I would go with something along the lines of "DON'T CHANGE THIS VALUE UNLESS YOU KNOW WHAT YOU ARE DOING".

Is that reasonable/sufficient or would you still like to keep immutability on that?